src/scopes/activityReport/reportText.js

```diff
@@ -1,9 +1,170 @@
+  return `
+  SELECT DISTINCT
+    "ActivityReports"."id"
+  FROM "ActivityReports"
+  LEFT JOIN "ActivityReportObjectives"
+  ON "ActivityReportObjectives"."activityReportId" = "ActivityReports"."id"
+  WHERE "ActivityReportObjectives"."ttaProvided"${a}`;
+};
+
+const activityReportResourcePosNeg = (pos = true) => {
+  const a = pos ? '' : ' IS NULL OR "Resources"."url"';
+
+  return `
+  SELECT DISTINCT
+    "ActivityReports"."id"
+  FROM "ActivityReports"
+  LEFT JOIN "ActivityReportResources"
+  ON "ActivityReportResources"."activityReportId" = "ActivityReports"."id"
+  LEFT JOIN "Resources"
+  ON "Resources"."id" = "ActivityReportResources"."resourceId"
+  WHERE "Resources"."url"${a}`;
+};
+
+const activityReportGoalResourcePosNeg = (pos = true) => {
+  const a = pos ? '' : ' IS NULL OR "Resources"."url"';
+
+  return `
+  SELECT DISTINCT
+    "ActivityReports"."id"
+  FROM "ActivityReports"
+  LEFT JOIN "ActivityReportGoals"
+  ON "ActivityReportGoals"."activityReportId" = "ActivityReports"."id"
+  LEFT JOIN "ActivityReportGoalResources"
+  ON "ActivityReportGoalResources"."activityReportGoalId" = "ActivityReportGoals"."id"
+  LEFT JOIN "Resources"
+  ON "Resources"."id" = "ActivityReportGoalResources"."resourceId"
+  WHERE "Resources"."url"${a}`;
+};
+
+const activityReportObjectiveResourcePosNeg = (pos = true) => {
+  const a = pos ? '' : ' IS NULL OR "Resources"."url"';
+
+  return `
+  SELECT DISTINCT
+    "ActivityReports"."id"
+  FROM "ActivityReports"
+  LEFT JOIN "ActivityReportObjectives"
+  ON "ActivityReportObjectives"."activityReportId" = "ActivityReports"."id"
+  LEFT JOIN "ActivityReportObjectiveResources"
+  ON "ActivityReportObjectiveResources"."activityReportObjectiveId" = "ActivityReportObjectives"."id"
+  LEFT JOIN "Resources"
+  ON "Resources"."id" = "ActivityReportObjectiveResources"."resourceId"
+  WHERE "Resources"."url"${a}`;
+};
+
+const nextStepsResourcePosNeg = (pos = true) => {
+  const a = pos ? '' : ' IS NULL OR "Resources"."url"';
+
+  return `
+  SELECT DISTINCT
+    "ActivityReports"."id"
+  FROM "ActivityReports"
+  LEFT JOIN "NextSteps"
+  ON "NextSteps"."activityReportId" = "ActivityReports"."id"
+  LEFT JOIN "NextStepResources"
+  ON "NextSteps"."id" = "NextStepResources"."nextStepId"
+  LEFT JOIN "Resources"
+  ON "Resources"."id" = "NextStepResources"."resourceId"
+  WHERE "Resources"."url"${a}`;
+};
+
+const activityReportContextPosNeg = (pos = true) => {
+  const a = pos ? '' : ' IS NULL OR "ActivityReports"."context"';
+
+  return `
+  SELECT DISTINCT
+    "ActivityReports"."id"
+  FROM "ActivityReports"
+  WHERE "ActivityReports"."context"${a}`;
+};
+
+const additionalNotesPosNeg = (pos = true) => {
+  const a = pos ? '' : ' IS NULL OR "ActivityReports"."additionalNotes"';
+
+  return `
+  SELECT DISTINCT
+    "ActivityReports"."id"
+  FROM "ActivityReports"
+  WHERE "ActivityReports"."additionalNotes"${a}`;
+};
+
+export function withReportText(searchText) {
+  const search = [`%${searchText}%`];
```

Comment:
do we need to run sequelize.escape on this?

Reply:
yeah, it happens here (filterAssociation -> reportInSubQuery):

https://github.com/HHS/Head-Start-TTADP/blob/10124f121963a9399f893da7d982aef6ae059b4b/src/scopes/activityReport/utils.js#L10